alternator/executor.cc

```diff
@@ -2873,6 +2876,46 @@ future<executor::request_return_type> executor::put_item(client_state& client_st
     co_return res;
 }
 
+future<executor::request_return_type> executor::execute_parsed_put_item(
+    client_state& client_state, 
+    tracing::trace_state_ptr trace_state, 
+    service_permit permit, 
+    put_item_lwt_args args) {
+    
+    _stats.api_operations.put_item++;
+    auto start_time = std::chrono::steady_clock::now();
+    
+    auto op = make_shared<put_item_operation>(*_parsed_expression_cache, _proxy, 
+                                              std::move(args.full_request));
+    
+    tracing::add_table_name(trace_state, op->schema()->ks_name(), op->schema()->cf_name());
+    const bool needs_read_before_write = op->needs_read_before_write();
+    
+    co_await verify_permission(_enforce_authorization, _warn_authorization, 
+                               client_state, op->schema(), auth::permission::MODIFY, _stats);
+    
+    auto cas_shard = op->shard_for_execute(needs_read_before_write);
+    
+    lw_shared_ptr<stats> per_table_stats = get_stats_from_schema(_proxy, *(op->schema()));
+    per_table_stats->api_operations.put_item++;
+    uint64_t wcu_total = 0;
+    
+    auto res = co_await op->execute(_proxy, std::move(cas_shard), client_state, 
+                                    trace_state, std::move(permit), needs_read_before_write, 
+                                    _stats, *per_table_stats, wcu_total);
+    
+    per_table_stats->operation_sizes.put_item_op_size_kb.add(
+        bytes_to_kb_ceil(op->consumed_capacity()._total_bytes));
+    per_table_stats->wcu_total[stats::wcu_types::PUT_ITEM] += wcu_total;
+    _stats.wcu_total[stats::wcu_types::PUT_ITEM] += wcu_total;
+    per_table_stats->api_operations.put_item_latency.mark(
+        std::chrono::steady_clock::now() - start_time);
+    _stats.api_operations.put_item_latency.mark(
+        std::chrono::steady_clock::now() - start_time);
+    
+    co_return res;
```

Comment:
This smells like it duplicates some existing code, could it be a helper function executed by both bounced and unbounced requests?